### PR TITLE
mach: Implement `LC_BUILD_VERSION`

### LINKED ISSUE #255: mach: Implement LC_BUILD_VERSION
Is there a reason `LC_BUILD_VERSION` isn't implemented yet? At first blush it seems straightforward and I'd be happy to work on making it happen.

### PR DESCRIPTION
**Pull request by messense**
Closes #255

### CODE REVIEW

File: src/mach/load_command.rs [hunk 1]
```diff
@@ -1269,6 +1298,19 @@ pub const LC_VERSION_MIN_TVOS: u32 = 0x2F;
...
+pub const PLATFORM_MACOS: u32 = 1;
+pub const PLATFORM_IOS: u32 = 2;
+pub const PLATFORM_TVOS: u32 = 3;
+pub const PLATFORM_WATCHOS: u32 = 4;
+pub const PLATFORM_BRIDGEOS: u32 = 5;
+pub const PLATFORM_MACCATALYST: u32 = 6;
+pub const PLATFORM_IOSSIMULATOR: u32 = 7;
+pub const PLATFORM_TVOSSIMULATOR: u32 = 8;
+pub const PLATFORM_WATCHOSSIMULATOR: u32 = 9;
+pub const PLATFORM_DRIVERKIT: u32 = 10;
```
Comment (messense) [inline src/mach/load_command.rs:1310]: `PLATFORM_*` consts comes from https://github.com/apple-oss-distributions/dyld/blob/c8a445f88f9fc1713db34674e79b00e30723e79d/common/MachOFile.h#L111-L124

Comment (m4b) [reply src/mach/load_command.rs:1310]: cool, thank you!

File: src/mach/load_command.rs [hunk 2]
```diff
@@ -1610,9 +1657,8 @@ impl<'a> ctx::TryFromCtx<'a, Endian> for CommandVariant {
                 let comm = bytes.pread_with::<LinkeditDataCommand>(0, le)?;
                 Ok((DyldChainedFixups(comm), size))
             }
-            // TODO: LC_NOTE (NoteCommand) and LC_BUILD_VERSION (BuildVersionCommand)
-            // are unimplemented.
-            LC_NOTE | LC_BUILD_VERSION | _ => Ok((Unimplemented(lc), size)),
```
Comment (m4b) [inline src/mach/load_command.rs:1615]: thanks for implementing this!

File: src/mach/load_command.rs [hunk 3]
```diff
@@ -1185,6 +1185,35 @@ pub struct EntryPointCommand {
...
+    /// X.Y.Z is encoded in nibbles xxxx.yy.zz
+    pub sdk: u32,
+    /// number of tool entries following this
+    pub ntools: u32,
+}
+
+/// Build tool version
+#[repr(C)]
+#[derive(Debug, Clone, Copy, Pread, Pwrite, IOread, IOwrite, SizeWith)]
+pub struct BuildToolVersion {
```
Comment (m4b) [inline src/mach/load_command.rs:1210]: I can't remember how this part works, is the user responsible for extracting these out from `BuildVersionCommand` `ntools` field? Or do we normally provide a method on `BuildVersionCommand` to extract related substructs?

**Review by m4b — APPROVED** [review]
This is great, thanks for implementing this!

**Comment by m4b** [comment]
Just had the one question about method, otherwise lgtm

File: src/mach/load_command.rs (hunk 3 above)
Comment (messense) [reply src/mach/load_command.rs:1210]: I think the user is responsible for extracting `BuildToolVersion`, the `object` crate didn't provide any helper functions either.

* https://docs.rs/object/latest/object/macho/struct.BuildVersionCommand.html
* https://docs.rs/object/latest/object/macho/struct.BuildToolVersion.html

**Comment by m4b** [comment]
thanks @messense ! do you need a point release for this soon or it's ok to wait a bit and collect some other PRs?

**Comment by messense** [comment]
It's ok to wait for a while.

**Comment by messense** [comment]
Hi @m4b, do you plan to release a new version soon? I'd like to use this for https://github.com/PyO3/maturin/issues/1252, thanks!

**Comment by m4b** [comment]
@messense sorry for delay life was busy, do you also need https://github.com/m4b/goblin/pull/342 ? I was considering branching and cherry-picking the non-breaking changes and delaying 342 until another breaking change? But if you need it can just release all at once in 0.7

**Comment by messense** [comment]
I dont need it.

**Comment by m4b** [comment]
@messense published all non-breaking changes as 0.6.1, and new 0.6.1 branch along with git tag, thanks for your patience!

**Pull request merged**